src/main.cpp

```diff
@@ -310,11 +322,26 @@ int main(int argc, char * argv[])
     if (!(ErrorDialogHandler::instance()->checkError())) {
         qDebug() << "Displaying mixxx";
         mixxx->show();
+        splash.finish(mixxx);
 
         qDebug() << "Running Mixxx";
         result = a.exec();
     }
 
+    QMessageBox cleaningUpBox;
+    cleaningUpBox.setStandardButtons(0);
+    cleaningUpBox.setWindowTitle(QObject::tr("Exiting Mixxx"));
+    cleaningUpBox.setText(QObject::tr(
+            "Cleaning up the library. This may take some time.\n"
+            "Do not restart Mixxx or shutdown the device until\n "
+            "this box has disappeared."));
```

Comment:
> In general, I feel that splash screens and shutdown screens are symptoms of problems and are not solutions.

You are right, that we have a bunch of related problems. But it is a common requirement, that a GUI application should not run too long without a GUI. 35 s start UP time is simply too long. 

If I got this right, Mac kills all applications with a launch time > 10 s 
http://stackoverflow.com/questions/26032085/xcode-6-process-launch-failed-timed-out-trying-to-launch-app 

I have read a study (lost the link) that a average user is willing to wait up to 5 s for an application to launch.  
The same is true for the shutdown. 

This PR Fixes all these timeout issues: Mixxx running = GUI available. 
Just for this topic, It is worth to commit this PR.